Cython/Utility/Overflow.c

```diff
@@ -170,17 +170,17 @@ static CYTHON_INLINE {{INT}} __Pyx_add_const_{{NAME}}_checking_overflow({{INT}}
     } else if (b < 0) {
         *overflow |= a < __PYX_MIN({{INT}}) - b;
     }
-    return a + b;
+    return ({{INT}}) ((unsigned {{INT}}) a + (unsigned {{INT}}) b);
```

Comment:
Could you at least add a comment here? It's not obvious from seeing the code that this is doing something reasonable.

I think referring to the SO question might be ok:
https://stackoverflow.com/questions/4240748/allowing-signed-integer-overflows-in-c-c